R/borrowing_class.R

```diff
@@ -1,43 +1,42 @@
-# class union ----
-setClassUnion("PriorOrNULL", c("Prior", "NULL"))
+setClassUnion(
+  "vectorOrNULL",
+  c("vector", "NULL")
+)
 
 #' `Borrowing` Class
 #'
 #' A class for defining borrowing details. Objects of class
-#' `Borrowing` should not be created directly but by the constructor
-#' [borrowing_details()].
+#' `Borrowing` should not be created directly but by the constructors
+#' [borrowing_hierarchical_commensurate()], [borrowing_none()], [borrowing_full()].
 #'
-#' @slot method character. The type of borrowing to perform. It
-#' must be one of: `'BDB'`, `'Full borrowing'`, or `'No borrowing'`. See `?borrowing_details` for
-#' more information.
-#' @slot ext_flag_col character. The name of the column in
-#' the data matrix that corresponds to the external control flag (`1`/`0` or
-#' `TRUE`/`FALSE`). This identifies a patient as belonging to the external
-#' control cohort.
-#' @slot tau_prior `Prior`. Object of class `Prior` defining the hyperprior on the
-#' "commensurability parameter". See `?borrowing_details` for more information.
 #' @slot data_stan_code string. Code to include in the Stan data program block.
-.borrowing_class <- setClass(
+#' @slot method_name string. The name of the method.
+#' @slot ext_flag_col character. Name of the external flag column in the matrix.
+#' @slot name_tau named vector for hierarchical commensurability parameter hyperprior.
+#' @family borrowing classes
+#' @seealso Prior constructor functions: [borrowing_full()], [borrowing_hierarchical_commensurate()], [borrowing_none()]
+setClass(
   "Borrowing",
   slots = c(
-    method = "character",
+    data_stan_code = "character",
+    method_name = "character",
     ext_flag_col = "character",
-    tau_prior = "PriorOrNULL",
-    data_stan_code = "character"
+    name_tau = "vectorOrNULL"
```

Comment:
I'm not sure about this in the parent class, since it's only needed for one specific borrowing model.